framework/include/utils/MooseTypes.h

```diff
@@ -943,39 +959,45 @@ struct enable_bitmask_operators<Moose::RelationshipManagerType>
     TheName(const char * s) : std::string(s) {}                                                    \
     TheName(size_t n, char c) : std::string(n, c) {}                                               \
   };                                                                                               \
-  namespace nlohmann                                                                               \
-  {                                                                                                \
-  template <>                                                                                      \
-  struct adl_serializer<TheName>                                                                   \
+  DerivativeStringToJSONDecl(TheName)
+
+/**
+ * This Macro is similar to DerivativeStringClass but additionally provides the implicit conversion
+ * operator to std::filesystem::path
+ */
+#define DerivativePathLikeStringClass(TheName)                                                     \
+  DerivativeStringClass(TheName##Base);                                                            \
+  class TheName : public TheName##Base                                                             \
   {                                                                                                \
-    static void to_json(json & j, const TheName & v);                                              \
+  public:                                                                                          \
+    using TheName##Base::TheName##Base;                                                            \
+    operator std::filesystem::path() const { return std::string(*this); }                          \
```

Comment:
:+1:

Reply:
Uhm, looks like this operator is not used for an implicit conversion though.

Reply:
fine...... I dropped this change.